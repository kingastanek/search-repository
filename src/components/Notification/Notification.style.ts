import styled from 'styled-components';

export const NotificationCloseButton = styled.button`
  background: none;
  border: none;
  cursor: pointer;
  height: 24px;
  outline: none;
  padding: 0;
`;

export const NotificationIconContainer = styled.div`
  display: flex;
  margin-right: 0.75rem;
`;

export const NotificationMessageContainer = styled.div`
  display: flex;
  font-size: 0.875rem;
  font-weight: 500;
  margin-right: 24px;
`;

export const NotificationWrapper = styled.div`
  background-color: var(--theme-white);
  border-radius: 8px;
  box-shadow: 0px 3px 6px #00000029;
  display: flex;
  line-height: 24px;
  max-width: 300px;
  padding: 20px 12px 20px 16px;
  position: fixed;
  top: 1rem;
  right: 1rem;
  z-index: 1000;
`;
