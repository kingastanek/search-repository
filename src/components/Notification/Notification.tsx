import React, { useEffect, useState } from 'react';
import Icon from 'components/Icon';
import {
  NotificationCloseButton,
  NotificationIconContainer,
  NotificationMessageContainer,
  NotificationWrapper,
} from './Notification.style';
import COLORS from 'styles/colors';

type tNotification = {
  message?: string | JSX.Element;
  show: boolean;
  type?: string;
  customClose?: boolean;
  closeNotification?: Function;
};

const Notification: React.FC<tNotification> = ({
  message = 'Something went wrong. Try again',
  show = false,
  type = 'error',
  customClose = false,
  closeNotification = () => {},
}): JSX.Element | null => {
  const [showNotification, setShowNotification] = useState(false);

  useEffect(() => {
    setShowNotification(show);
  }, [show]);

  const closeNotifications = () => {
    setShowNotification(false);
    if (customClose) {
      closeNotification(false);
    }
  };

  const getIconType = (notificationType: string) => {
    switch (notificationType) {
      case 'error':
        return 'error';
      case 'info':
        return 'info';
      case 'success':
        return 'checkCircleOutline';
      default:
        return 'error';
    }
  };

  const getIconFill = (notificationType: string) => {
    switch (notificationType) {
      case 'error':
        return COLORS.RED;
      case 'info':
        return COLORS.BLUE;
      case 'success':
        return COLORS.GREEN;
      default:
        return COLORS.RED;
    }
  };

  if (!showNotification) return null;

  return (
    <NotificationWrapper>
      <NotificationMessageContainer>
        <NotificationIconContainer>
          <Icon fill={getIconFill(type)} icon={getIconType(type)} />
        </NotificationIconContainer>
        {message}
      </NotificationMessageContainer>
      <NotificationCloseButton onClick={closeNotifications}>
        <Icon fill={COLORS.RED} icon='close' />
      </NotificationCloseButton>
    </NotificationWrapper>
  );
};

export default Notification;
