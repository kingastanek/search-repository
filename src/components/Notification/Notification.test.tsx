import React from 'react';
import { shallow } from 'enzyme';
import Notification from './Notification';

describe('Notification', () => {
  const baseProps = {
    show: false,
  };

  const notificationDefault = shallow(<Notification {...baseProps} />);

  it('should be hidden', () => {
    expect(notificationDefault).toMatchSnapshot();
  });
});
