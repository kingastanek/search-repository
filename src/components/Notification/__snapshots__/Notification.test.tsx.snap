// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should be hidden 1`] = `ShallowWrapper {}`;
